server/telemetry.go

```diff
@@ -3,65 +3,49 @@ package main
 import (
 	"strings"
 
+	"github.com/mattermost/mattermost-plugin-api/experimental/bot/logger"
+	"github.com/mattermost/mattermost-plugin-api/experimental/telemetry"
 	"github.com/mattermost/mattermost-server/v5/model"
-	analytics "github.com/segmentio/analytics-go/v2"
 )
 
 const (
 	NPS_FEEDBACK = "nps_feedback"
 	NPS_SCORE    = "nps_score"
-
-	SEGMENT_KEY = "5xaDYWpjOoCKmJNNKK6fg1DacwZ7ZVZc"
 )
 
 func (p *Plugin) initializeClient() error {
-	client := analytics.New(SEGMENT_KEY)
-
-	if !p.blockSegmentEvents {
-		if err := client.Identify(&analytics.Identify{
-			UserId: p.API.GetDiagnosticId(),
-		}); err != nil {
-			return err
-		}
-	}
+	client, err := telemetry.NewRudderClient()
 
 	p.client = client
+	return err
```

Comment:
It's a bit surprising to see `p.client` getting set even if `err != nil`. I was expecting something like:

```suggestion
	client, err := telemetry.NewRudderClient()
	if err != nil {
		return err
	}
	
	p.client = client
	
	return nil
```

Reply:
The Tracker, if it receives a null client, just decides not to track. I felt that was the best way to deal with these issues.
Probably needs documentation.

Reply:
Got it, thanks for the explanation. I still fells weird to me to have a tracker that can't track because a client is missing, but that not a blocking concern.

Reply:
This allow to keep the code clean. If not, we should be having all over the place stuff like "If tracker != nil {tracker.track}".

Reply:
Also, if the EnableDiagnostics value is false, we also have a tracker that cannot track :sweat_smile: